traveltimepy/mapper.py

```diff
@@ -0,0 +1,510 @@
+                )
+            ],
+            intersections=[]
+        )
+    elif departure_time is not None:
+        return TimeMapRequest(
+            departure_searches=[
+                time_map.DepartureSearch(
+                    id=f'Search {ind}',
+                    coords=cur_coordinates,
+                    travel_time=travel_time,
+                    departure_time=departure_time,
+                    transportation=transportation,
+                    range=search_range
+                )
+                for ind, cur_coordinates in enumerate(coordinates)
+            ],
+            arrival_searches=[],
+            unions=[
+                time_map.Union(
+                    id='Union search',
+                    search_ids=[f'Search {ind}' for ind, _ in enumerate(coordinates)]
+                )
+            ],
+            intersections=[]
+        )
+    else:
+        raise ApiError('arrival_time or departure_time should be specified')
+
+
+def create_routes(
+    locations: List[Location],
+    searches: Dict[LocationId, List[LocationId]],
+    transportation: Union[PublicTransport, Driving, Ferry, Walking, Cycling, DrivingTrain],
+    departure_time: Optional[datetime],
+    arrival_time: Optional[datetime],
+    properties: Optional[List[Property]],
+    full_range: Optional[FullRange]
+) -> RoutesRequest:
+    if properties is None:
+        properties = [Property.TRAVEL_TIME]
+    if arrival_time is not None:
+        return RoutesRequest(
+            locations=locations,
+            arrival_searches=[
+                routes.ArrivalSearch(
+                    id=arrival_id,
+                    arrival_location_id=arrival_id,
+                    departure_location_ids=[departure_id for departure_id in departure_ids],
+                    arrival_time=arrival_time,
+                    transportation=transportation,
+                    properties=properties,
+                    full_range=full_range
+                )
+                for arrival_id, departure_ids in searches.items()
+            ],
+            departure_searches=[]
+        )
+    elif departure_time is not None:
+        return RoutesRequest(
+            locations=locations,
+            departure_searches=[
+                routes.DepartureSearch(
+                    id=departure_id,
+                    departure_location_id=departure_id,
+                    arrival_location_ids=[arrival_id for arrival_id in arrival_ids],
+                    departure_time=departure_time,
+                    transportation=transportation,
+                    properties=properties,
+                    full_range=full_range
+                )
+                for departure_id, arrival_ids in searches.items()
+            ],
+            arrival_searches=[]
+        )
+    else:
+        raise ApiError('arrival_time or departure_time should be specified')
+
+
+def create_proto_request(
```

Comment:
Coming from other languages  I would prefer this file split into multiple route based one (Or something similar)
If for whatever reasons this is preferred way of doing things in python, you can ignore my comment

Reply:
yeah, I'll think about it. I think that I can make it better.